src/derivative_wrappers.jl

```diff
@@ -40,55 +40,104 @@ function jacobian(f, x, integrator)
     alg = unwrap_alg(integrator, true)
     local tmp
     if alg_autodiff(alg)
-      J = jacobian_autodiff(f, x)
-      tmp = 1
+      if DiffEqBase.has_colorvec(integrator.f)
+        J,tmp = jacobian_autodiff(f, x, integrator)
+      else
+        J,tmp = jacobian_autodiff(f, x)
+      end
     else
-      J = jacobian_finitediff(f, x, alg.diff_type, integrator)
-      N = length(x)
-      if alg.diff_type==Val{:complex} && eltype(x)<:Real
-        tmp = N
-      elseif alg.diff_type==Val{:forward}
-        tmp = N + 1
+      if DiffEqBase.has_colorvec(integrator.f)
+        J,tmp = jacobian_finitediff(f, x, alg.diff_type, integrator, integrator.f.colorvec)
       else
-        tmp = 2N
+        J,tmp = jacobian_finitediff(f, x, alg.diff_type, integrator)
       end
     end
     integrator.destats.nf += tmp
     J
 end
 
-jacobian_autodiff(f, x) = ForwardDiff.derivative(f,x)
-jacobian_autodiff(f, x::AbstractArray) = ForwardDiff.jacobian(f, x)
+jacobian_autodiff(f, x) = (ForwardDiff.derivative(f,x),1)
+jacobian_autodiff(f, x::AbstractArray) = (ForwardDiff.jacobian(f, x),1)
+function jacobian_autodiff(f, x::AbstractArray, integrator)
```

Comment:
Maybe we could to do the function evaluation counting properly, e.g. `nf = length(u) \div chunk_size`. That could be a separate PR.